### ISSUE TITLE
Lane connector: Different source node shape for monorail

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#295]
Would it be possible to change source node shape for monorail tracks?

Currently they use circles, like road lanes. I initially thought there was a routable lane on the median (as I'd never even tried routing monorails before) or something.

Maybe if the monorail source nodes were square or rectangular? I realise this would add some complexity in terms of orientation (ensure square/rectangle is aligned with lane) but I think it would give a much more intuitive interface to the end-user.

Tagging: #282

**Comment by krzychu124** [comment]
If I manage to create that first shader I mentioned in #282 I could create other shapes like pentagon, hexagon etc... It's pure math problem written in [Cg](https://en.wikipedia.org/wiki/Cg_(programming_language))

**Comment by aubergine10** [comment]
If it's any use there are some shader tutorials here: https://catlikecoding.com/unity/tutorials/

**Comment by krzychu124** [comment]
Yeah I know that site 😉

**Comment by krzychu124** [comment]
In addition, maybe we should add some sort of switch for cycling through lane connection modes - (preview or manage connections):

Available views/states:
- road vehicles,
- rail vehicles,
- both

**Comment by aubergine10** [comment]
Yup - dependent on discussion starting at https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/issues/38#issuecomment-516898701 (ie. where to show context-specific tool modes/options)

**Comment by kianzarrin** [comment]
Mixed case: Some assets use the same lane for both car/tram like the vanilla two way road with tram tracks.
![mixed](https://user-images.githubusercontent.com/26344691/74612230-04310400-510c-11ea-8346-0f14b894cf7d.png)

Separate Case: Some other custom assets have tram lane and bus/car lane on the top of each other.
![separate](https://user-images.githubusercontent.com/1386719/58762230-08010400-8546-11e9-9762-9af3b65496db.jpg)

Shapes: #354
- cars lane: Circle 
- trams/monorail lanes: rectangle
- tram on top of car lane: rectangle on top of circle? triangle?
- mixed tram/car lane: rectangle on top of circle? triangle?
- bus lane : bundle it with car lane, give it different shape?
- monorail on top of tram lane( see #654)  : this is theoretically possible. Its and edge case that may be visually annoying but does not prevent the user from selecting monorail and tram/car under separately. 

User control:
- ALT to switch to track mode?
- hotkey (eg: CTRL+M) to switch between modes: 
option 1: mixed, car | mono/tram track (only if there are monorail or tram tracks)
option 2: mixed | car | tram | monorail | bus. (skip mode if unavailable on the selected node)

back-end:
its pretty easy to know what to do in the case when tram and bus/car lane are separated.
What should we do if tram and car lane is mixed? 

In many places cars can enter bus lane only if they want to turn. should we have possibility for bus to go straight but car to turn in lane connector?

**Comment by aubergine10** [comment]
Separating bus and car lane would be bad idea having given it some thought. To user, it's just "a bus lane".

The distinction is really about "something that has tires (car, bus, truck...)" vs. "someting that runs on tracks (tram, monorail, train...)".

> Note: there are some crazy assets with things like metro track or rail track within a road.

I think maybe either `Ctrl` or `Alt` to toggle between road vs. tracks mode. (Obviously on road only prefab it would have no effect. Likewise, on track only prefab, it would have no effect.)

Then that leaves the issue that there could be more than one track in the same "position", for example a monorail track over a tram track. I'm not sure what to do about that situation without making everything really confusing. The lane connectors could perhaps be placed side by side, but even then how would we highlight the specific track they relate to?

**Comment by kianzarrin** [comment]
@aubergine10 Are we going to make it impossible to let cars use bus lane to turn? I know this question is opening up a whole new can of worms and leads to complexity and results into many combinations of possibilities so we need to be careful about how we go through with this. maybe have it on your mind and we can come back to this in future?

I opened issue #716

**Comment by kvakvs** [comment]
If all tools are soon going to have their floating tool window, you can plan here what will that window contain?
* Keyboard shortcuts help? (Delete hint)
* Stay in lane button with Ctrl+S hint
* A row of switches for traffic types?
Do not bother with actual layout or sprites, just the controls that will be there.

**Comment by aubergine10** [comment]
> Are we going to make it impossible to let cars use bus lane to turn?

No, of course not.

To the user, a bus lane is just a road lane that's painted red. If they want to keep cars and trucks off that lane, they enable the 'ban cars and trucks from bus lanes' option (or whatver it is called) in mod options.

We would never have per-vehicle-category lane connectors on road lanes.

The distiction here is _only_ about tracked vehicles (trains, trams, etc) vs. tire vehicles (cars, trucks, etc).

Bus lane = it is just a normal road lane from perspective of the lane connector tool.

**Comment by aubergine10** [comment]
Suggestion:

* We have a shortcut (whatever it is) to toggle between "road mode" and "track mode"
* We worry about the issue of "tracks above tracks" as a separate problem some time in the future
